sound/soc/sof/topology.c

```diff
@@ -2107,8 +2107,8 @@ static int sof_set_pipe_widget(struct snd_sof_dev *sdev, struct snd_sof_widget *
 			}
 	}
 
-	/* set the pipe_widget and apply the dynamic_pipeline_widget_flag */
-	swidget->pipe_widget = pipe_widget;
+	/* set the pipeline and apply the dynamic_pipeline_widget_flag */
+	swidget->spipe = spipe;
```

Comment:
doesh't this conflict with the assignment here (previous patch) https://github.com/thesofproject/linux/pull/3957/commits/841a7d0ee0bb206142b4d2da767205b6c196e16b#diff-15e8b91662eaff37556ce9a2418e6824c0ad644bab20b40da1a0a5085195a5cdR1552

Reply:
no it wont. this sets the spipe for all other widgets. I've added a special check to skip scheduler widgets now to clarify the confusion.